### DISCUSSION TITLE
How to map <Esc> in kitty + vim

### DISCUSSION THREAD

**Discussion by skysky97** [kovidgoyal/kitty#6277] [Q&A] [answered]
`map <Esc> :close<CR>` works in gnome-terminal + vim, but not in kitty + vim.

When I press CTRL-V + ESC, it shows `^[[27u`, and `map ^[[27u :close<CR>` works in kitty + vim, but not in gnome-terminal.

What's the right way to map <Esc>.

**Comment by kovidgoyal** [comment, marked as answer]
You would need to ask the vim developers why mapping ESC is not working.
Most likely you just need to update vim, vim's initial implementation of
the kitty keyboard protocol was very buggy.

**Reply by skysky97** [reply]
Update vim to latest version 9.0.1563 dose solve this problem.
